tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp

```diff
@@ -452,6 +452,22 @@ class AstNamedCastExpr : public AstExpr {
   }
 };
 
+template <typename T, class Function>
+void DumpList(T start, T end, AstDumper* dumper, Function function)
```

Comment:
I'd declare `function` as `std::function<void(AstDumper*, decltype(*start)&>)>`:
```cpp
#include <iostream>
#include <vector>

#include <functional>

template<typename T>
void f(T iter, std::function<void(decltype(*iter)&)> fn)
{
    fn(*iter);
}

int main()
{
    int i = 1;
    int* p = &i;
    f(p, [&](auto& val) { val = 2; });
    std::cout << *p << "\n";

    std::vector<int> vec = { 3 };
    f(vec.begin(), [&](auto& val) { val = 4; });
    std::cout << vec[0];
}
```

Reply:
Hmmm. I thought about that, but wasn't quite sure if that gave me enough flexibility. Let me poke around and see what comes out.

Reply:
I just thought, `std::function<void(AstDumper* dumper, decltype(*start)& element)> doDumpElement` could possibly be a better name if you still have that function around (I can't find it in the latest PR iteration.